tests/assert-jobs/reporter-otlp/main.go

```diff
@@ -92,74 +89,42 @@ func initProvider(serviceName string) func() {
 
 	logrus.Debugln("Using", reportingProtocol, "to report the traces")
 
-	var driver otlp.ProtocolDriver
-
+	var exp *otlptrace.Exporter
+	var err error
 	switch reportingProtocol {
 	case "grpc":
-		driver = otlpgrpc.NewDriver(
-			otlpgrpc.WithInsecure(),
-			otlpgrpc.WithEndpoint(collector),
-			otlpgrpc.WithDialOption(grpc.WithBlock()),
+		exp, err = otlptracegrpc.New(ctx,
+			otlptracegrpc.WithInsecure(),
+			otlptracegrpc.WithEndpoint(collector),
+			otlptracegrpc.WithDialOption(grpc.WithBlock()),
 		)
 	case "http":
-		driver = otlphttp.NewDriver(
-			otlphttp.WithInsecure(),
-			otlphttp.WithEndpoint(collector),
+		exp, err = otlptracehttp.New(ctx,
+			otlptracehttp.WithInsecure(),
+			otlptracehttp.WithEndpoint(collector),
 		)
 	default:
 		logrus.Fatalln("Reporting protocol", reportingProtocol, "not recognized")
 	}
 
-	exp, err := otlp.NewExporter(ctx, driver)
-	if err != nil {
-		logrus.Fatalln("error creating the exporter", err)
-	}
-
-	res, err := resource.New(ctx,
-		resource.WithAttributes(
-			semconv.ServiceNameKey.String(serviceName),
-		),
-	)
 	if err != nil {
-		logrus.Fatalln("error creating the resource", err)
+		logrus.Fatalln("error creating", reportingProtocol, "exporter", err)
 	}
 
-	bsp := sdktrace.NewBatchSpanProcessor(exp)
 	tracerProvider := sdktrace.NewTracerProvider(
 		sdktrace.WithSampler(sdktrace.AlwaysSample()),
-		sdktrace.WithResource(res),
-		sdktrace.WithSpanProcessor(bsp),
-	)
-
-	cont := controller.New(
-		processor.New(
-			simple.NewWithExactDistribution(),
-			exp,
-		),
-		controller.WithExporter(exp),
-		controller.WithCollectPeriod(2*time.Second),
```

Comment:
what I said [her](https://github.com/jaegertracing/jaeger-operator/pull/2084#discussion_r990806809)e also applies to these changes.